### PR TITLE
Export delete file

### PR DESCRIPTION
**Pull request by iilyak**
In order to support `rename_on_delete` semantics for views. We want to reuse `couch_server:delete_file` in couch_mrview_cleanup.

### CODE REVIEW

**Comment by rnewson** [comment]
+1, and nice cleanup of a cut-and-paste in the test. The assumption that the rename takes less than one second predates your change but it would be nice to get rid of it.

File: src/couch_server.erl [hunk 1]
```diff
@@ -549,9 +554,12 @@ delete_db_file(RootDir, FullFilePath, Options) ->
     end.
 
 rename_on_delete(Original) ->
+    file:rename(Original, renamed_filename(Original)).
+
+renamed_filename(Original) ->
```
Comment (kxepal) [inline src/couch_server.erl]: It's actually "deleted_filename", unless you make `deleted` suffix customizable.

File: src/couch_server.erl [hunk 2]
```diff
@@ -549,9 +554,13 @@ delete_db_file(RootDir, FullFilePath, Options) ->
     end.
 
 rename_on_delete(Original) ->
+    file:rename(Original, deleted_filename(Original)).
+
+deleted_filename(Original) ->
     {{Y,Mon,D}, {H,Min,S}} = calendar:universal_time(),
+    UUID = ?b2l(couch_uuids:random()),
```
Comment (rnewson) [inline src/couch_server.erl]: I get what you intended but this seems awkward. Could delete_file instead return `ok | {renamed, NewName} | {error, Reason}` instead? Then we know where to look in the tests.

Comment (iilyak) [inline src/couch_server.erl]: Good idea. I'll change it to return:
- `{ok, deleted}`
- `{ok, {renamed, FileName}}`
- `{error, Reason}`

File: src/couch_server.erl [hunk 3]
```diff
@@ -538,20 +540,31 @@ db_closed(Server, Options) ->
...
 rename_on_delete(Original) ->
+    DeletedFileName = deleted_filename(Original),
+    case file:rename(Original, DeletedFileName) of
+        ok -> {ok, {renamed, DeletedFileName}};
+        Else -> Else
+    end.
+
+deleted_filename(Original) ->
     {{Y,Mon,D}, {H,Min,S}} = calendar:universal_time(),
+    UUID = ?b2l(couch_uuids:random()),
```
Comment (rnewson) [inline src/couch_server.erl]: no need for the uuid, timestamp suffices

**Comment by rnewson** [comment]
+1 after reverting the uuid change. the original names were fine for uniqueness, the issue was the assumption that the test runs within one second (it would fail spuriously if the clock ticked between some steps).

**Comment by iilyak** [comment]
I'll revert the UUID change and squash for final review.

**Comment by rnewson** [comment]
+1

**Pull request merged**